src/App.tsx

```diff
@@ -1,25 +1,63 @@
-// import React from 'react';
+import {
+    useState,
+    FormEvent,
+    ChangeEvent
+} from 'react';
+import { debounce } from 'lodash';
 import './App.css';
 import {
-    QueryType,
-    useData
+    QueryType
 } from './useData';
+import { StockChart } from './stockChart';
 
 function App() {
-    const data = useData({
+    const [ query, setQuery ] = useState({
         queryType: QueryType.RANGE,
-        queryString: '15/02/2022-15/05/2022'
+        queryString: ''
     });
+    const [ leftRange, setLeftRange ] = useState('');
+    const [ rightRange, setRightRange ] = useState('');
+
+    function handleSubmit(ev: FormEvent<HTMLFormElement>) {
+        ev.preventDefault();
+        setQuery({
+            queryType: QueryType.RANGE,
+            queryString: `${leftRange}-${rightRange}`
+        });
+    }
+
+    const handleLeftRange = debounce(
+        (ev: ChangeEvent<HTMLInputElement>) => setLeftRange(ev.target.value), 400);
+
+    const handleRightRange = debounce(
+        (ev: ChangeEvent<HTMLInputElement>) => setRightRange(ev.target.value), 400);
 
     return (
         <div className="App">
             <header className="App-header">
-                BH Homework
+                {`BH Homework displaying chart for range: "${query.queryString}"`}
             </header>
             <main>
-                {
-                    data.map((item) => <div>{item.date}</div>)
-                }
+                <div>
+                    <form onSubmit={handleSubmit}>
+                        <input
+                            type="text"
+                            placeholder="dd/mm/yyyy"
+                            required={true}
+                            onChange={(ev) => handleLeftRange(ev)}
+                            pattern={'[0-9]{2}/[0-9]{2}/[0-9]{4}'}
+                        /> - 
```

Comment:
To be perfectly honest I would normally use a 3rd party date selector, but I really wanted to pattern for a very naive client side navigation. Refreshing knowldeg about debounce was also a good thing.